FirebaseTestProject/ViewModels/PostData.swift

```diff
@@ -25,4 +54,43 @@ import Foundation
             print(error)
         }
     }
+    
+    func loadFavorites() async {
+        do {
+            let posts = try await PostService.getFavorites()
+            self.posts = posts
+        }
+        catch {
+            print(error)
+        }
+    }
+    
+    func index(of post: Post) -> Int? {
+        for i in posts.indices {
+            if posts[i].id == post.id {
+                return i
+            }
+        }
+        return nil
+    }
+    
+    // Returns if Succeeded
+    func remove(post: Post) -> Bool {
+        // Returns if element not found
+        guard let index = posts.firstIndex(where: { $0.id == post.id }) else { return false }
+        posts.remove(at: index)
+        Task {
+            try await PostService.delete(post: post)
+        }
+        // We can return true here because we only want to know if the index exists
+        return true
+    }
+}
+
+extension PostData {
+    enum DataType {
```

Comment:
`DataType` is a bit inexact to capture these cases

Reply:
Removed this - Completed